pytorch_lightning/core/lightning.py

```diff
@@ -308,6 +315,8 @@ def log(
                 each dataloader to not mix values
             batch_size: Current batch_size. This will be directly inferred from the loaded batch,
                 but some data structures might need to explicitly provide it.
+            attribute_name: This would be the attribute name to find a given metric.
+                If None, this will be automatically inferred by Lightning.
```

Comment:
Can you give an example? It is not clear to me why this would be different of how pytorch determines the name for `nn.Modules`. As the metrics are just `nn.Modules`, I would assume loading and saving state dict works as expected.

Reply:
It is the same as PyTorch way to find the `prefix_name` for a given nn.Module.
I am using this prefix to find back the reference on the LightningModule on ResultCollection state_dict reload.